region/lowernorfair/west/Screw Attack Room.json

```diff
@@ -968,45 +1456,207 @@
         "leaveWithGrappleTeleport": {
           "blockPositions": [[2, 34]]
         }
-      }
+      }
+    },
+    {
+      "id": 38,
+      "link": [3, 2],
+      "name": "Reset Fall Speed (Wave)",
+      "requires": [
+        {"obstaclesCleared": ["A"]},
+        {"doorUnlockedAtNode": 2},
+        "canResetFallSpeed",
+        "Wave",
+        {"heatFrames": 200}
+      ],
+      "unlocksDoors": [
+        {
+          "types": ["ammo"],
+          "requires": [
+            "never"
+          ]
+        }
+      ]
+    },
+    {
+      "id": 39,
+      "link": [3, 2],
+      "name": "Reset Fall Speed (Quick Shot)",
+      "requires": [
+        {"obstaclesCleared": ["A"]},
+        {"doorUnlockedAtNode": 2},
+        "canResetFallSpeed",
+        "canTrickyJump",
+        {"heatFrames": 145}
+      ],
+      "unlocksDoors": [
+        {
+          "types": ["super"],
+          "requires": [
+            "canPrepareForNextRoom"
+          ]
+        },
+        {
+          "types": ["missiles", "powerbomb"],
+          "requires": [
+            "never"
+          ]
+        }
+      ]
+    },
+    {
+      "link": [3, 2],
+      "name": "Come In Shinecharging, Leave Shinecharged (Wall Jump)",
+      "entranceCondition": {
+        "comeInShinecharging": {
+          "length": 3,
+          "openEnd": 1
+        }
+      },
+      "requires": [
+        "canShinechargeMovementTricky",
+        "canWalljump",
+        {"heatFrames": 150}
+      ],
+      "exitCondition": {
+        "leaveShinecharged": {
+          "framesRemaining": 35
+        }
+      },
+      "unlocksDoors": [
+        {
+          "types": ["super"],
+          "requires": []
+        },
+        {
+          "types": ["missiles", "powerbomb"],
+          "requires": [
+            "never"
+          ]
+        }
+      ],
+      "note": [
+        "Press down precisely to gain the shinecharge while sliding off the ledge."
+      ]
+    },
+    {
+      "link": [3, 2],
+      "name": "Come In Shinecharging, Leave With Spark",
+      "entranceCondition": {
+        "comeInShinecharging": {
+          "length": 3,
+          "openEnd": 1
+        }
+      },
+      "requires": [
+        "canShinechargeMovementComplex",
```

Comment:
The difficulty depends on your run speed.  Its mostly a problem because you are coming through a door.  Continuing to fall to B is ShinechargeMovementTricky, but I find this 1 tile break to be only annoying.

Reply:
Yeah I saw it's possible to break the one block without sliding off the ledge: https://videos.maprando.com/video/1484. Makes sense that it should be easier at lower run speeds, since you would have more frames where you would be in contact with the bomb block before falling off. I think collision oscillation might play a role too, not sure. Do you think something needs to change in the requirements here?

Reply:
It would only be situations where you have forced high run speed entering the room like leaving SpeedBoosterHall. Otherwise the leniency should allow you to control your speed well enough.  So I don't think its a concern.

Leaving SpeedBoosterHall with full room run speed is in Hard with HeroShot, when it should be Very hard right2left and either VH or EXP left2right.